### PR TITLE
 Create integration test suite and migrate appropriate tests from unit test suite #579

### PR DESCRIPTION
**Pull request by erangeles**

### CODE REVIEW

**Comment by erangeles** [comment]
the failing build is due to other component's failed tests which this issue does not address.

**Comment by notmessenger** [comment]
@joshforisha @juwara0 @azizpunjani What do we think of the `'"Options" property needs to be an object'` and `'"Series" property needs to be an array'` tests remaining unit tests?  I think I'm leaning towards making them integration tests but could be convinced of either.

File: tests/integration/components/sl-chart-test.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent( 'sl-chart', 'Integration | Component | sl chart', {
+    integration: true
+});
+
+test( 'Defaults applied correctly', function( assert ) {
+    this.set( 'testseries', []);
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    this.set( 'testoptions', {});
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    );
+
+    assert.ok(
+      this.$( '.sl-chart' ).hasClass( 'sl-panel' ),
+      'has class sl-panel'
+    );
+});
+
+test( 'Loading state adds loading class', function( assert ) {
+    this.set( 'testseries', []);
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    this.set( 'testoptions', {});
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+});
+
+test( 'Defaults applied correctly', function( assert ) {
+    this.set( 'testseries', []);
+    this.set( 'testoptions', {});
+
+    this.render( hbs`
+      {{#sl-chart series=testseries options=testoptions}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+});
+
+test( 'Loading state adds loading class', function( assert ) {
+    this.set( 'testseries', []);
+    this.set( 'testoptions', {});
+
+    this.render( hbs`
+      {{#sl-chart series=testseries isLoading=false options=testoptions}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    assert.strictEqual(
+        this.$( '.sl-chart' ).hasClass( 'sl-loading' ),
+        false,
+        'Default rendered component does not have "sl-loading" class'
+    );
+
+    this.render( hbs`
+      {{#sl-chart series=testseries isLoading=true options=testoptions}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+    assert.strictEqual(
+        this.$( '.sl-chart' ).hasClass( 'sl-loading' ),
+        true,
+        'Default rendered component does have "sl-loading" class'
+    );
+});
+
+test( 'Title property is set', function( assert ) {
+    this.set( 'testseries', []);
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    this.set( 'testoptions', {});
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+
+    this.render( hbs`
+      {{#sl-chart series=testseries options=testoptions}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    assert.strictEqual(
+        this.$( '.panel-heading' )[0],
+        undefined,
+        'No chart title is rendered when title is not set'
+    );
+
+    this.render( hbs`
+      {{#sl-chart title='Peak serve hours' series=testseries options=testoptions}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+    this.set( 'testseries', []);
+    this.set( 'testoptions', {});
+
+    this.render( hbs`
+      {{#sl-chart series=testseries options=testoptions}}
+      {{/sl-chart}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '.panel-heading' )[0],
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: Why is this accessed via `[0]` whereas the assertion further down can be accessed via `html()` without the array index?

File: tests/integration/components/sl-chart-test.js [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+    assert.equal(
+        this.$( '.panel-heading' ).html(),
+        'Peak serve hours',
+        'Chart title is created with title value'
+    );
+});
+
+test( 'Chart div uses the correct style', function( assert ) {
+    this.set( 'testseries', []);
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+    this.set( 'testoptions', {});
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+    this.render( hbs`
+      {{#sl-chart series=testseries options=testoptions}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+    assert.equal(
+        this.$( 'div.chart' ).attr( 'style' ),
+        'height: auto; width: auto;',
+        'Chart div has automatic height and width'
+    );
+
+    this.render( hbs`
+      {{#sl-chart series=testseries options=testoptions height=10 width=20}}
+      {{/sl-chart}}
+    ` );
```
Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/ember-style-guide/blob/master/javascript.md#block-statements

File: tests/integration/components/sl-chart-test.js (hunk 5 above)
Comment (erangeles) [inline tests/integration/components/sl-chart-test.js]: https://github.com/softlayer/sl-ember-components/pull/669/files So I went off how @juwara0 made his spacing in his render statements in that PR. Not exactly sure how this is wrong.

Comment (notmessenger) [inline tests/integration/components/sl-chart-test.js]: Please have a conversation with @juwara0 and @azizpunjani, and@joshforisha, as https://github.com/softlayer/sl-ember-components/pull/647 introduced the pattern that I directed you to change this PRs changes to, and I assumed that it was the desired/correct one, as it passed the linting.

Within that PR, https://github.com/softlayer/sl-ember-components/blob/EmberCli-1.13.8/tests/integration/components/sl-modal-body-test.js#L9 also introduced an inconsistency within/against itself that I missed that needs to be brought into alignment with whatever is decided.

Comment (juwara0) [inline tests/integration/components/sl-chart-test.js]: Aziz is out of the office this week and next week but before he left I mentioned to him that his tests were failing linting and we agreed that he needed to update his block statements to follow the convention that I set. Aziz mentioned that the would submit a PR to make those fixes but I suspect he ran out of time before and did not get it in before he left for vacation.

**Comment by notmessenger** [comment]
Any input @juwara0 @joshforisha on https://github.com/softlayer/sl-ember-components/pull/674#issuecomment-132870866 ?

**Comment by joshforisha** [comment]
@notmessenger I think those should probably be unit tests. They would only check how a method on the component handles a value, unrelated to how the element is rendered (directly) or used in a broader sense.

**Comment by erangeles** [comment]
concur with @joshforisha point.

**Comment by juwara0** [comment]
@notmessenger - My thought is that we should keep the type checking series of tests as unit tests since that level of testing is more aligned with concept of a unit test in which we would check methods, parameters, returns from methods and logic of methods.

**Pull request merged**